src/events/kyuubi.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Kyuubi related event handlers."""
+
+import ops
+from ops import CharmBase
+
+from core.context import Context
+from events.base import BaseEventHandler, compute_status
+from managers.kyuubi import KyuubiManager
+from utils.logging import WithLogging
+from workload.base import KyuubiWorkloadBase
+
+
+class KyuubiEvents(BaseEventHandler, WithLogging):
+    """Class implementing S3 Integration event hooks."""
+
+    def __init__(self, charm: CharmBase, context: Context, workload: KyuubiWorkloadBase):
+        super().__init__(charm, "kyuubi")
+
+        self.charm = charm
+        self.context = context
+        self.workload = workload
+
+        self.kyuubi = KyuubiManager(self.workload)
+
+        self.framework.observe(self.charm.on.install, self._on_install)
+        self.framework.observe(self.charm.on.kyuubi_pebble_ready, self._on_kyuubi_pebble_ready)
+        self.framework.observe(self.charm.on.update_status, self._update_event)
+        self.framework.observe(self.charm.on.config_changed, self._on_config_changed)
+
+    @compute_status
+    def _on_install(self, event: ops.InstallEvent) -> None:
```

Comment:
The decorator `compute_status` will take care of setting appropriate status at the end of hook, that's why the function definition is not needed.